### PR TITLE
Migrate to esbuild

### PR DESCRIPTION
**Pull request by iddan**
- Replace Gulp and Babel with esbuild
 - Replace commonjs imports with ECMAModules imports wherever possible

### CODE REVIEW

**Comment by ide** [comment]
Before we can merge this, we need a test plan to make sure this works in Expo apps. Probably the simplest way to do this is to create a new Expo project, add your local copy of this repo as a dependency, and then render a simple component like the one in this repo's README. Also, since there are changes to the font-related code, the test plan should render fonts as well.

**Comment by iddan** [comment]
How should we execute the test plan? Should it be a project that can be ran manually or is there a go-to way to run in it in the CI (I couldn't find resources about it)

**Comment by ide** [comment]
It doesn't need to be automated necessarily. Making a project manually and testing it is fine. Just want to make sure things still work.

**Comment by iddan** [comment]
As requested a test repository: https://github.com/iddan/expo-2d-context-test

**Review by ide — COMMENTED** [review]
The fonts look like they could potentially break. Make sure to test that too.

File: timesnewroman.js [hunk 1]
```diff
@@ -1397,8 +1399,8 @@ kerning first=1077 second=1091 amount=-2
 kerning first=932 second=963 amount=-5
 kerning first=160 second=87 amount=-1
   `,
-  {"timesnewroman1.png":require("./assets/fonts/timesnewroman1.png"),
-   "timesnewroman2.png":require("./assets/fonts/timesnewroman2.png")},
+  {"timesnewroman1.png": timesnewroman1,
+   "timesnewroman2.png": timesnewroman2},
```
[the remark this answers is not in the document]
Comment (iddan) [reply timesnewroman.js]: I recovered the code for the fonts files. And adding a test in the test repository.

[the remark this answers is not in the document]
Comment (iddan) [reply timesnewroman.js]: Yes

**Comment by iddan** [comment]
Last week I added tests for fonts, can we move forward with the PR?

**Review by ide — APPROVED** [review]

**Pull request merged**